src/components/MovieForm/MovieForm.test.js

```diff
@@ -0,0 +1,151 @@
+  });
+
+  test('should not mount when isVisible false', () => {
+    render(<MovieForm info={{}} genreList={['Comedy']} isVisible={false} />);
+    expect(screen.queryByTestId('MovieForm')).not.toBeInTheDocument();
+  });
+
+  test('should mount Add Movie when info.id is falsy', () => {
+    render(<MovieForm info={{}} genreList={['Comedy']} isVisible={true} />);
+    expect(screen.queryByText('Add Movie')).toBeInTheDocument();
+  });
+
+  test('should mount Edit Movie when info.id is truthy', () => {
+    render(<MovieForm info={{ id: 1 }} genreList={['Comedy']} isVisible={true} />);
+    expect(screen.queryByText('Edit Movie')).toBeInTheDocument();
+  });
+
+  test('should leave form fields empty when Add Movie', () => {
+    render(<MovieForm info={{}} genreList={['Comedy']} isVisible={true} />);
+
+    expect(screen.getByText('Release Date')).not.toHaveValue();
+    expect(screen.getByTestId('movie-form-title')).not.toHaveValue();
+    expect(screen.getByTestId('movie-form-image-url')).not.toHaveValue();
+    expect(screen.getByTestId('movie-form-genre')).not.toHaveValue();
+    expect(screen.getByTestId('movie-form-runtime')).not.toHaveValue();
+    expect(screen.getByTestId('movie-form-overview')).not.toHaveValue();
+  });
+
+  test('should fill form fields with values from info', () => {
+    const info = {
+      id: 1,
+      title: 'Test Title',
+      releaseDate: '2014-01-13',
+      imageUrl: 'https://image.png',
+      genre: ['Drama'],
+      runtime: '2h',
+      overview: 'very interesting movie',
+    };
+    render(<MovieForm info={info} genreList={['Comedy', 'Drama']} isVisible={true} />);
+
+    let formControl = within(screen.getByTestId('movie-form-title'));
+    expect(formControl.getByRole('textbox')).toHaveValue('Test Title');
+    expect(screen.getByPlaceholderText('MM / DD / YYYY')).toHaveValue('01 / 13 / 2014');
+
+    formControl = within(screen.getByTestId('movie-form-image-url'));
+    expect(formControl.getByRole('textbox')).toHaveValue('https://image.png');
+
+    formControl = within(screen.getByTestId('movie-form-genre'));
+    expect(formControl.getByText('Drama')).toBeInTheDocument();
+
+    formControl = within(screen.getByTestId('movie-form-runtime'));
+    expect(formControl.getByRole('textbox')).toHaveValue('2h');
+
+    formControl = within(screen.getByTestId('movie-form-overview'));
+    expect(formControl.getByRole('textbox')).toHaveValue('very interesting movie');
+  });
+
+  test('should reset form fields when reset button is clicked', async () => {
+    const info = {
+      id: 1,
+      title: 'Test Title',
+      rating: 7.1,
+      releaseDate: '2014-01-13',
+      imageUrl: 'Test Url',
+      genre: ['Drama'],
+      runtime: '2h',
+      overview: 'Test Overview',
+    };
+    render(<MovieForm info={info} genreList={['Comedy', 'Drama']} isVisible={true} />);
+
+    let formControl = within(screen.getByTestId('movie-form-title'));
```

Comment:
I think it's not a good approach to use the same variable always,
this can be more readable
```suggestion
    const titleField = within(screen.getByTestId('movie-form-title'));
```